### PR TITLE
fix(batch): reject explicit zero offer limits

### LINKED ISSUE #4249: batch-runner: `--limit 0` means unlimited, which is the opposite of what a caller passing 0 means
**Where:** `batch/batch-runner.sh`, `LIMIT=0` default and `if (( LIMIT > 0 ))` guards (currently around lines 53, 1389, 1461).

**What happens:** `--limit N` caps the pass at N offers, except N=0, which is the internal sentinel for "no limit". A wrapper or cron job that computes a limit and arrives at 0 (memory pressure, a dry run, a "scan only" pass) launches an unbounded batch instead of none. In one real case a scheduled run passed 0, queued 62 offers and spent roughly 25 minutes of model time on 8 of them before it was killed.

**Why it is worth changing:** the flag reads as a count; 0 as "all" is only knowable from the source. Every other place a user meets a count (`--parallel`, `--max-retries`) treats 0 literally or rejects it.

**Suggested fix (any of):**
1. Treat an explicit `--limit 0` as "process nothing" and print `Limit: 0 (nothing to do)`; keep the unset default as unlimited.
2. Or reject `--limit 0` with a one-line error pointing at "omit the flag for no limit".
3. At minimum, print a loud `Limit: none (0 = unlimited)` in the banner so a log reader sees what is about to happen.

Happy to send the PR for whichever you prefer; it is a few lines either way. Noting it as an issue first because it changes documented behaviour.

### PR DESCRIPTION
**Pull request by luochen211**
## What does this PR do?

Reject an explicit `--limit 0` before checking worker prerequisites or touching batch state and locks. Previously it silently selected the unlimited sentinel and could process the entire queue. The error directs callers to omit `--limit` for an unlimited run; positive limits and the unset default keep their existing behavior.

Normalize leading zeroes as decimal and reject values above Bash's signed 64-bit maximum before arithmetic can turn a cap into an unlimited run. Validate missing and malformed limit arguments at the same boundary and document the positive-integer contract in CLI help and batch mode instructions. Regression tests execute the actual runner in isolated fixtures, checking early rejection, lock preservation, maximum/overflow boundaries, leading-zero decimal inputs, and the pending queue for omitted and positive limits.

## Related issue

Fixes #4249 (the explicit-rejection option proposed in the issue).

## Type of change

- [x] Bug fix

## Validation

- `node --test tests/batch-runner-limit.test.mjs`
- `node test-all.mjs`: 9,205 passed, 0 failed (optional live Gemini smoke test skipped without a key)
- `npm run lint`
- `git diff --check`

## Checklist

- [x] Read CONTRIBUTING.md.
- [x] No personal data or user-layer changes.
- [x] Bug fix on the existing batch execution path.

## Summary

`--limit` now accepts only positive integers in `batch/batch-runner.sh:123-128`.

Explicit `--limit 0`, malformed values, negative values, overflowing values, and missing values fail before batch state or locks change. The error directs users to omit `--limit` for an unlimited run.

Positive limits retain their behavior. Omitted `--limit` remains unlimited.

## User impact

- Unlimited run: omit `--limit`.
- Limited run: use a positive integer, such as `--limit 10`.
- Zero is no longer treated as unlimited.
- Invalid input leaves queue state and the PID lock unchanged.

## Files changed

- `batch/batch-runner.sh`
- `modes/batch.md`
- `tests/batch-runner-limit.test.mjs`

`AGENTS.md`, `update-system.mjs`, `DATA_CONTRACT.md`, `providers/`, and `.github/` were not changed.

### CODE REVIEW

File: modes/batch.md [hunk 1]
```diff
@@ -104,7 +104,7 @@ Options:
 - `--retry-failed` — retry only failed jobs
 - `--resume-paused` — resume jobs paused after a Claude session/rate limit
 - `--start-from N` — start from ID N
-- `--limit N` — max number of jobs to process in this run
+- `--limit N` — max number of jobs to process in this run (positive integer; `0` is rejected, omit the flag for no limit)
```
[the remark this answers is not in the document]
Comment (luochen211) [reply modes/batch.md:107]: The core CLI help and modes/batch.md now describe the positive-only contract. I checked the linked documentation repository: its CONTRIBUTING.md explicitly disallows documenting behavior that exists only in an open core PR (https://github.com/career-ops-hq/career-ops-docs/blob/main/CONTRIBUTING.md#what-we-do-not-accept-without-talking-first). The current public examples use positive limits and remain valid. The FAQ/glossary/guide clarification should follow after this core change is merged, rather than publish unshipped behavior now.

File: batch/batch-runner.sh [hunk 2]
```diff
@@ -120,7 +120,14 @@ while [[ $# -gt 0 ]]; do
     --retry-failed) RETRY_FAILED=true; shift ;;
     --resume-paused) RESUME_PAUSED=true; shift ;;
     --start-from) START_FROM="$2"; shift 2 ;;
-    --limit) LIMIT="$2"; shift 2 ;;
+    --limit)
+      if [[ $# -lt 2 ]] || ! [[ "$2" =~ ^[0-9]+$ ]] || [[ "$2" =~ ^0+$ ]]; then
```
[the remark this answers is not in the document]
Comment (luochen211) [reply batch/batch-runner.sh:124]: Fixed in 57acde2. The parser strips leading zeroes before arithmetic and rejects values above 9223372036854775807 with a length/lexical check, so overflow cannot reach the queue. Tests execute the real runner at the maximum, the first overflowing value, a wrap-to-zero value, a padded overflow, and a leading-zero decimal value (008). Full suite: 9,205 passed, 0 failed; lint passed.

**Review by santifer — APPROVED** [review]

**Comment by santifer** [comment]
Merged. Thank you, @luochen211: **`--limit 0` now stops with a clear error before the runner touches the lock or the queue**, instead of silently meaning "no limit" and processing everything. Covering negatives, decimals, empty values and overflow in the test closes the edge that was left open when `--limit` first landed.
